python/pyspark/sql/types.py

```diff
@@ -1616,6 +1616,12 @@ def to_arrow_type(dt):
         if type(dt.elementType) == TimestampType:
             raise TypeError("Unsupported type in conversion to Arrow: " + str(dt))
         arrow_type = pa.list_(to_arrow_type(dt.elementType))
+    elif type(dt) == StructType:
+        if any(type(field.dataType) == StructType for field in dt):
+            raise TypeError("Nested StructType not supported in conversion to Arrow")
```

Comment:
Is ArrayType(elementType = StructType) supported?

Reply:
Good catch, that should not be supported right now. I added a check and put that type in a test.

Reply:
I am curious why `ArrayType(elementType = StructType)` support  was removed from here? @BryanCutler

Reply:
@cfmcgrady support wasn't removed, it was never allowed to have `ArrayType(elementType = StructType)` - I don't think there was an explicit check before this. It might be possible to add this in the future, but it's a little tricky to represent this in Pandas efficiently, I believe.

Reply:
Thank you for your reply.